File: 800/116A/main.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
    int n;
    cin>>n;
    
    int maxi = INT_MIN;
    int pass = 0;
    
    for(int i=0;i<n;i++)
    {
        int a,b;
        cin>>a>>b;
        
        maxi = max(maxi, pass-a+b);
        pass = pass-a+b;
    }
    
    cout<<maxi;
    return 0;
}

File: 800/116A/main.java
import java.util.*;

public class main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();   // number of stops
        int maxi = Integer.MIN_VALUE; // maximum passengers at any time
        int pass = 0;           // current passengers

        for (int i = 0; i < n; i++) {
            int a = sc.nextInt(); // passengers leaving
            int b = sc.nextInt(); // passengers entering

            maxi = Math.max(maxi, pass - a + b);
            pass = pass - a + b;
        }

        System.out.println(maxi);
    }
}
